src/main/kotlin/de/bund/digitalservice/useid/config/ContentSecurityPolicyProperties.kt

```diff
@@ -0,0 +1,32 @@
+package de.bund.digitalservice.useid.config
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.stereotype.Component
+import org.springframework.validation.annotation.Validated
+import javax.validation.constraints.NotBlank
+import javax.validation.constraints.NotEmpty
+
+@Component
+@ConfigurationProperties(prefix = "csp")
+@Validated
+class ContentSecurityPolicyProperties {
+    @NotBlank
+    lateinit var defaultConfig: String
+
+    @NotBlank
+    lateinit var frameAncestors: String
+
+    @NotEmpty
+    lateinit var allowedHosts: List<String>
+
+    fun domainIsAllowed(host: String): Boolean {
+        return allowedHosts.contains(host)
+    }
+    fun getCSPHeaderValue(host: String): String {
+        return "$defaultConfig$frameAncestors $host;"
```

Comment:
Even less code duplication ;) 

```suggestion
        return "${getDefaultCSPHeaderValue()} $host;"
```

Reply:
we can add it because it will return `some default value;frame-ancestors 'self'; foo.bar;` the semicolon from default value is included, which should not happen.